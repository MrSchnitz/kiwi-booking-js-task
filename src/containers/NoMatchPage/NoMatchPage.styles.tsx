import styled from "styled-components";

export const NoMatchPageWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;

  width: 100%;
  min-height: 100vh;
`;
