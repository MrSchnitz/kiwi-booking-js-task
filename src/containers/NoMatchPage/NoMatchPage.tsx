import React from "react";
import { NoMatchPageWrapper } from "./NoMatchPage.styles";
import {
  Button,
  Heading,
  Illustration,
  Stack,
} from "@kiwicom/orbit-components";
import { Link } from "react-router-dom";

export default function NoMatchPage() {
  return (
    <NoMatchPageWrapper>
      <Stack flex={true} direction="column" justify="center" align="center">
        <Heading type="display">Sorry, nothing to see here!</Heading>
        <Illustration name="Error404" />
        <Link to="/">
          <Button>Return to homepage</Button>
        </Link>
      </Stack>
    </NoMatchPageWrapper>
  );
}
