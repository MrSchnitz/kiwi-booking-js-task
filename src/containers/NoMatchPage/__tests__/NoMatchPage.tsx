import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NoMatchPage from "../NoMatchPage";

describe("NoMatchPage", () => {
  it("should render basic component", () => {
    const { container } = render(
      <MemoryRouter>
        <NoMatchPage />
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
