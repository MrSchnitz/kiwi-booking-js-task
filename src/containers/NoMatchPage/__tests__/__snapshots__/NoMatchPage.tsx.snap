// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatchPage should render basic component 1`] = `
<div>
  <div
    class="sc-bczRLJ hfDjeB"
  >
    <div
      class="Stack__StyledStack-sc-oaff2v-0 eWjCkv"
    >
      <div
        class="Heading__StyledHeading-sc-rrd5d-0 joCkMC"
      >
        Sorry, nothing to see here!
      </div>
      <img
        alt="Error404"
        class="IllustrationPrimitive__StyledImage-sc-2hh260-0 lkLFfm"
        src="//images.kiwi.com/illustrations/0x200/Error404-Q85.png"
        srcset="//images.kiwi.com/illustrations/0x400/Error404-Q85.png 2x, //images.kiwi.com/illustrations/0x600/Error404-Q85.png 3x"
      />
      <a
        href="/"
      >
        <button
          class="ButtonPrimitive__StyledButtonPrimitive-sc-1lbd19y-0 iRvWIY"
          type="button"
        >
          <div
            class="ButtonPrimitiveContent__StyledButtonPrimitiveContent-sc-1r81o9a-0 brimGn"
          >
            <div
              class="ButtonPrimitiveContentChildren__StyledButtonPrimitiveContentChildren-sc-1m4y8u8-0 jaHPWv"
            >
              Return to homepage
            </div>
          </div>
        </button>
      </a>
    </div>
  </div>
</div>
`;
